apps/labs/posts/pep-517-build-system-popularity.md

```diff
@@ -0,0 +1,615 @@
+`setup.cfg`](https://setuptools.pypa.io/en/latest/userguide/declarative_config.html)
+was added in 30.3.0. It supports most of the common configuration options
+but not all. For example, Python extensions written in C can only be
+declared via `setup.py`. However, it also supports some new features,
+such as automatically extracting the version from a Python file or reading
+the package description from a README file:
+
+```ini
+[metadata]
+version = attr: frobnicate.__version__
+long_description = file: README.rst
+```
+
+Finally, support for [`pyproject.toml`
+configuration](https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html)
+was added in 61.0.0. It is based on [PEP
+621](https://peps.python.org/pep-0621/), and therefore makes the common
+part of the configuration compatible with other PEP 517 build backends
+such as Hatchling.
+
+<div style={{textAlign: 'center'}}>
+
+<table id="table-3" style={{width: 'auto', margin: '0 2em', display: 'inline-block', verticalAlign: 'top'}}>
+  <caption>Table 3. Counts for setuptools configuration format combinations</caption>
+  <tr><th>Formats</th><th>Count</th></tr>
+  <tr><td>`setup.py`</td><td align='right'>3750</td></tr>
+  <tr><td>`setup.cfg` + `setup.py`</td><td align='right'>1104</td></tr>
+  <tr><td>`pyproject.toml`</td><td align='right'>541</td></tr>
+  <tr><td>`pyproject.toml` + `setup.py`</td><td align='right'>330</td></tr>
+  <tr><td>`setup.cfg`</td><td align='right'>87</td></tr>
+  <tr><td>`pyproject.toml` + `setup.cfg` + `setup.py`</td><td align='right'>17</td></tr>
+  <tr><td>`pyproject.toml` + `setup.cfg`</td><td align='right'>14</td></tr>
+  <tr><td>(no configuration — broken distribution)</td><td align='right'>11</td></tr>
+</table>
+<table id="table-4" style={{width: 'auto', margin: '0 2em', display: 'inline-block', verticalAlign: 'top'}}>
+  <caption>Table 4. Cumulative counts for every configuration format</caption>
+  <tr><th>Format</th><th>Total</th></tr>
+  <tr><td>(all packages)</td><td align='right'>5854</td></tr>
+  <tr><td>`setup.py`</td><td align='right'>5201</td></tr>
+  <tr><td>`setup.cfg`</td><td align='right'>1222</td></tr>
+  <tr><td>`pyproject.toml`</td><td align='right'>902</td></tr>
+</table>
+
+</div>
+
+<figure id="fig-3" style={{float: 'right', maxWidth: '372px', padding: '0 1em'}}>
+  <img src="/posts/pep-517-build-system-popularity/fig3.svg"
+    alt="A Venn Diagram of setuptools configuration formats. It consists of
+    three overlapping circles: one big circle representing setup.py use
+    and two smaller circles representing setup.cfg and pyproject.toml.
+    The setup.cfg circle is mostly overlapping with setup.py, while
+    the pyproject.toml circle is closer to half-overlapped."
+  />
+
+  <figcaption>Figure 3. Use of setuptools configuration formats combinations</figcaption>
+</figure>
+
+`setup.py` still remains the most popular of the configuration formats.
+It is used by 89% of the analyzed packages, with 64% not using any other
```

Comment:
```suggestion
It is used by 89% of the analyzed setuptools-using packages, with 64% not using any other
```

or something similar. just to avoid the "that's larger than all packages using setuptools together" response